File: test/sysv_posix_comp/posix_2.py
import posix_ipc
import os
import time

# Connect to existing
sem = posix_ipc.Semaphore("/capstone")


for x in range(10):
    sem.acquire()
    print("Critical! posix 2")
    time.sleep(2)
    print("Leaving! posix 2")
    sem.release()
    time.sleep(1)

#sem.unlink()
sem.close()

File: test/sysv_posix_comp/posix_2.cpp
#include <iostream>

// Named Semaphore headers
// linke with -lpthread
#include <fcntl.h>
#include <sys/stat.h>
#include <semaphore.h>

// exit()
#include <stdlib.h>
// sleep()
#include <unistd.h>

int main()
{
  // Named Semaphore
  sem_t * sem = sem_open("/capstone", O_CREAT, 0666, 1);
  if (sem == SEM_FAILED) {
    std::cout<<"*** *** sem_open failed!!!"<<std::endl;
    sem_unlink("/capstone");
    exit(1);
  }
  else {
    for (int i = 0; i < 10; ++i) {
      sem_wait(sem);
      std::cout<<"Critical! posix 2"<<std::endl;
      sleep(2);
      std::cout<<"Leaving!  posix 2"<<std::endl;
      sem_post(sem);
      sleep(1);
    }

    sem_close(sem);
  }
}
